Formula/mingw-w64.rb

```diff
@@ -0,0 +1,177 @@
+          system "make", "install"
+        end
+        info.rmtree
+        (share/"locale").rmtree
+      end
+      ENV.prepend_path "PATH", bin.to_s
+
+      args = %W[
+        --host=#{target_arch}
+        --prefix=#{prefix}/#{target_arch}
+      ]
+      args << "--disable-multilib" if target_arch.start_with?("i686") || (target_arch.start_with?("x86_64") && build.without?("multilib"))
+
+      mkdir "mingw-w64-headers/build-#{target_arch}" do
+        system "../configure", *args
+        system "make"
+        system "make", "install"
+      end
+
+      ln_s "#{prefix}/#{target_arch}", "#{prefix}/mingw"
+
+      resource("gcc").stage buildpath/"gcc"
+
+      gcc_args = %W[
+        --target=#{target_arch}
+        --prefix=#{prefix}
+        --with-sysroot=#{prefix}
+        --enable-version-specific-runtime-libs
+        --with-bugurl=https://github.com/Homebrew/homebrew-core/issues
+        --enable-languages=c,c++,fortran
+        --with-ld=#{bin}/#{target_arch}-ld
+        --with-as=#{bin}/#{target_arch}-as
+        --with-gmp=#{Formula["gmp"].opt_prefix}
+        --with-mpfr=#{Formula["mpfr"].opt_prefix}
+        --with-mpc=#{Formula["libmpc"].opt_prefix}
+        --with-isl=#{Formula["isl"].opt_prefix}
+        MAKEINFO=#{Formula["texinfo"].opt_bin}/makeinfo
+      ]
+      if target_arch.start_with?("i686")
+        gcc_args << "--disable-multilib"
+      elsif target_arch.start_with?("x86_64") && build.without?("multilib")
+        gcc_args << "--enable-64bit" << "--disable-multilib"
+      elsif target_arch.start_with?("x86_64") && build.with?("multilib")
+        gcc_args << "--enable-targets=all"
+      end
+
+      mkdir "#{buildpath}/gcc/build-#{target_arch}" do
+        system "../configure", *gcc_args
+        system "make", "all-gcc"
+        system "make", "check-gcc"
+        system "make", "install-gcc"
+      end
+
+      ENV.prepend_path "PATH", bin.to_s
+
+      args = %W[
+        CC=#{target_arch}-gcc
+        CXX=#{target_arch}-g++
+        CPP=#{target_arch}-cpp
+        LD=#{target_arch}-gcc
+        --host=#{target_arch}
+        --prefix=#{prefix}/#{target_arch}
+        --with-sysroot=#{prefix}/#{target_arch}
+      ]
+
+      if target_arch.start_with?("i686")
+        args << "--enable-lib32" << "--disable-lib64"
+      elsif target_arch.start_with?("x86_64") && build.without?("multilib")
+        args << "--disable-lib32" << "--enable-lib64"
+      elsif target_arch.start_with?("x86_64") && build.with?("multilib")
+        args << "--enable-lib32" << "--enable-lib64"
+      end
+
+      mkdir "mingw-w64-crt/build-#{target_arch}" do
+        system "../configure", *args
+        system "make"
+        system "make", "install"
+      end
+
+      chdir "#{buildpath}/gcc/build-#{target_arch}" do
+        system "make"
+        system "make", "install"
+      end
+
+      ln_s "../../lib/gcc/#{target_arch}/lib/libgcc_s.a", "#{prefix}/#{target_arch}/lib"
+
+      ENV["LDPATH"] = "#{target_arch}/#{lib}"
```

Comment:
It may be working, but I think LDPATH should be for host linker, not target linker. An alternative is `--with-ld=` and `--with-as` (see: https://gist.github.com/fxcoudert/538c3d1ff2e55a348018a00e03d2f156) which I think helps it work in multilib (I got it working with the formula in that link).